recipe/bld.bat

```diff
@@ -0,0 +1,5 @@
+set LIBRARY_DIRS="%LIBRARY_BIN%;%LIBRARY_LIB%"
+set INCLUDE_DIRS="%LIBRARY_INC%"
```

Comment:
@fhs I'm probably missing something obvious or I have a mistake in the patch. These line should be equivalent to what you have in the AppVeyor build link you sent. Can you see anything I may have done wrong here?

Reply:
You have `set "hdf4=4.2"` and pyhdf is getting confused thinking `4.2` is the directory where you have hdf4 installed. Try upsetting that environment variable.

Reply:
Sorry, I'm not sure what you meant. The only vars set are those two above. You mean I should do:

```bat
set HDF4=4.2
```

?

Reply:
No, I meant you *shouldn't* do that. Maybe you can just remove the lines where you set the `hdf4` environment variable in these files:
```
.ci_support/win_c_compilervs2008python2.7.yaml
.ci_support/win_c_compilervs2015python3.6.yaml
.ci_support/win_c_compilervs2015python3.7.yaml
```

Reply:
I see those files are autogenerated, so maybe you need to  unset it:
```
set hdf4=
```

Reply:
Ah OK. Let me see why those are in there. Thanks!

Reply:
OK. That is the version pin in the yaml file and we need that for the build but I can probably unset in the `bld.bat`, not 100% sure though b/c it may mess with the final pinning scheme.

Reply:
As an innocent bystander, @fhs are you saying that these settings in the YAML file are setting an environment variable named `HDF4` that is conflicting with an environment variable that `pyhdf` expects to use to point to the path of the HDF4 C library?

Reply:
@djhoese Yes, although `HDF4` environment variable is entirely optional. There is no standard place hdf4 libraries are installed (sadly), so pyhdf has to look around environment variables or setup.py arguments to find it.

Reply:
That is not a problem and we can circumvent that. (I'm actually surprised this is the first time it happened!)

However, it seems that we are still having trouble funding the libs in the conda-build staged. I'll try to get a Windows machine to debug this.

Reply:
Looks like the path was not fully expanded to the build prefix, it is failing with:

```
    Running command python setup.py egg_info
    ******
    "%PREFIX%\Library\include" not found
    ******
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-cov_z1p9\setup.py", line 130, in <module>
        raise RuntimeError(msg)
    RuntimeError: Cannot proceed without the HDF4 library.  Please export INCLUDE_DIRS and LIBRARY_DIRS as explainedin the INSTALL file.
```

`%PREFIX%` should have been expanded there :-/

Reply:
There are quotes around the path for some reason. Maybe remove the quotes when you're setting the environment variables:
```
set LIBRARY_DIRS=%LIBRARY_BIN%;%LIBRARY_LIB%
set INCLUDE_DIRS=%LIBRARY_INC%
```
or do this:
```
set "LIBRARY_DIRS=%LIBRARY_BIN%;%LIBRARY_LIB%"
set "INCLUDE_DIRS=%LIBRARY_INC%"
```

Reply:
Thanks @fhs. Windows quotes are still a mystery to me :smile:

Python 2.7 seems to be passing :tada: 

Not sure what is going on with Python 3 yet...